### PR TITLE
fix IndexError for layer names with double underscores

### PR DESCRIPTION
**Pull request by thawn**
fixes issue #33

### CODE REVIEW

**Comment by thawn** [comment]
finally managed 😓 
sorry for the confusion.

File: src/napari_workflows/_tests/test_function.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def refine(image:napari.types.LabelsData) -> napari.types.LabelsData:
     gui = make_gui(segment, viewer, auto_call=True)
     viewer.window.add_dock_widget(gui)
     # invoke execution / workflow update
-    gui.sigma.value = gui.sigma.value
+    gui.sigma.value = 5.1
```
Comment (thawn) [inline src/napari_workflows/_tests/test_function.py:60]: apparently, the  workflow update is only triggered if the value actually changes

File: src/napari_workflows/_tests/test_function.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def refine(image:napari.types.LabelsData) -> napari.types.LabelsData:
     # test code generation
     code = manager.to_python_code()
     print(code)
-    assert len(code.split("\n")) == 19
+    assert len(code.split("\n")) == 16
```
Comment (thawn) [inline src/napari_workflows/_tests/test_function.py:102]: The code that was generated on my system looks fine to me. However, it has only 16 lines, not 19. @haesleinhuepf please verify that this code is correct:
```python
from skimage.io import imread
import napari_workflows  # version 0.2.8
import napari
if 'viewer' not in globals():
    viewer = napari.Viewer()

# segment

image1_s = napari_workflows.segment(image0_I, 2)
viewer.add_labels(image1_s, name='Result of segment')

# refine

image2_r = napari_workflows.refine(image1_s)
viewer.add_labels(image2_r, name='Result of refine')
```

**Comment by haesleinhuepf** [comment]
Awesome, the fix works like a charm. Also thanks for the detailed bug report @thawn  ! 1+

**Pull request merged**